sklearn/ensemble/_hist_gradient_boosting/_binning.pyx

```diff
@@ -64,7 +64,8 @@ cdef void _map_num_col_to_bins(const X_DTYPE_C [:] data,
             # for known values, use binary search
             left, right = 0, binning_thresholds.shape[0]
             while left < right:
-                middle = (right + left - 1) // 2
+                # take the difference to prevent overflowing
```

Comment:
I'm not sure this comment really helps without context. If we want to add a comment, I'd suggest something like

> equal to (right + left - 1) // 2 but avoids overflow